sycl/include/sycl/vector_preview.hpp

```diff
@@ -1293,6 +1011,7 @@ template <typename Type, int NumElements> class vec {
   }
 #endif // __SYCL_USE_EXT_VECTOR_TYPE__
 
+public:
```

Comment:
`setValue` and `getValue` should not be public members. We risk users using them.

Reply:
`setValue` and `getValue` are used by the math operators, which are now friends of `vec_arith` class. I'm not sure how to allow math operators access `getValue` and `setValue` without making them public.
In my follow-up PR to change storage type of `sycl::vec` to `std::array` on device (https://github.com/intel/llvm/pull/13270/files#diff-f4893f252981b17787572acdeb04d8c48d63219330a191b7b251d96bce8ac82a), I have gotten rid of all getters and setters in math operations, so, I can make these `getValue`/`setValue` private again once my changes go through (hopefully soon).

Reply:
Interesting. I am not sure I like having an in between where we suddenly expose `setValue` and `getValue`. You patch could change and we could forget to remove/hide `getValue` and `setValue`.

My usual go-to in these situations is to define a "helper" function or class in the `detail` namespace, which has the same interface (in this case `getVaue` and `setValue`), make it a friend on the corresponding class, and then have everything that needs `getValue` and `setValue` do it through that. Users _can_ get access to that, but since it is in the detail namespace we wash our hands.

Reply:
> Interesting. I am not sure I like having an in between where we suddenly expose `setValue` and `getValue`. You patch could change and we could forget to remove/hide `getValue` and `setValue`.
> 
> My usual go-to in these situations is to define a "helper" function or class in the `detail` namespace, which has the same interface (in this case `getVaue` and `setValue`), make it a friend on the corresponding class, and then have everything that needs `getValue` and `setValue` do it through that. Users _can_ get access to that, but since it is in the detail namespace we wash our hands.

That's a nice idea! I can try that.

Reply:
Fixed in 19aa68d0ed45f1d7c2e94772031368b84881d257